components/03-organisms/site-footer/yds-site-footer.twig

```diff
@@ -16,61 +16,163 @@
   'data-component-width': 'site',
   'data-site-footer-border-thickness': site_footer__border_thickness|default('0'),
   'data-footer-theme': site_footer__theme|default('one'),
+  'data-footer-variation': site_footer__variation|default('basic'),
   class: bem(site_footer__base_class),
 } %}
 
 <footer {{ add_attributes(site_footer__attributes) }}>
   <div {{ bem('inner', [], site_footer__base_class) }}>
-    {# Primary #}
-    <div {{ bem('primary', [], site_footer__base_class) }}>
-      {# Branding #}
-      <div {{ bem('branding', [], site_footer__base_class) }}>
+
+    {% if site_footer__variation == 'mega' %}
```

Comment:
@joetower All of the code looks good. My question is more about organization. This shows an overloaded component that can be basic or mega. Unlike the menu, these two variations have different content needs. Different inputs with different outputs. Should they be different components within the footer? 

On the other hand, the basic footer and the mega footer will never be reused in other places. So I'm not sure it makes sense to break them out into separate compoents. This just feels a little incorrect to me as I try to avoid lots of big-chunks of markup in a condition within a template. I used to see this in WordPress all of the time and the code would become fragile as you would trace the if/elseif/else controls.

No changes needed. Just sharing my first impression.